cookbooks/redis/recipes/install_from_source.rb

```diff
@@ -0,0 +1,29 @@
+redis_version = node['redis']['version']
+redis_download_url = node['redis']['download_url']
+redis_installer_directory = '/opt/redis-source'
+
+remote_file "/opt/redis-#{redis_version}.tar.gz" do
+  source "#{redis_download_url}"
+  owner node[:owner_name]
+  group node[:owner_name]
+  mode 0644
+  backup 0
+end
+
+execute "unarchive Redis installer" do
+  cwd "/opt"
+  command "tar zxf redis-#{redis_version}.tar.gz && sync"
+end
+
+execute "Remove old redis-source" do
```

Comment:
Is this needed because redis tarball has the same directory? It's probably bettero to extract to `/opt/redis-<version>` . This might causes headaches debugging things in production.